### PR TITLE
Title: Adding mstlink to mstflint

### PR DESCRIPTION
**Pull request by Mustafadall**
Signed-off-by: Mustafa Dalloul <mustafadall@mellanox.com>

### CODE REVIEW

File: mlxlink/modules/mlxlink_user_input.h [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#include <stdint.h>
+#include <mlxreg/mlxreg_lib.h>
+#include <mlxreg/mlxreg_parser.h>
+
+using namespace std;
+
+class UserInput {
+public:
+    UserInput();
+    ~UserInput()
```
Comment (samerd) [inline mlxlink/modules/mlxlink_user_input.h]: virtual ~UserInput();

File: mlxlink/modules/mlxlink_utils.h [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+#if __cplusplus <= 199711L
+template<typename T>
+std::string to_string(T toConvert)
+{
+    std::stringstream out;
+    out << toConvert;
+    return out.str();
+}
+#endif
+void printFlagLine(const char flag_s, const string &flag_l, const string &param,
```
Comment (samerd) [inline mlxlink/modules/mlxlink_utils.h]: move all print functions to printutil

File: mlxlink/modules/printutil/mlxlink_cmd_print.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+#define MLXLINK_CMD_PRINT_H
+
+#include <vector>
+#include "mlxlink_record.h"
+
+class MlxlinkCmdPrint{
+public:
+    MlxlinkCmdPrint();
+    virtual ~MlxlinkCmdPrint();
+    friend std::ostream & operator << (std::ostream &out, const MlxlinkCmdPrint &c);
```
Comment (samerd) [inline mlxlink/modules/printutil/mlxlink_cmd_print.h]: give meaningful name: const MlxlinkCmdPrint &c

**Comment by Mustafadall** [comment]
> Thanks Mustafa,
> looks great.
> please see comments below.
> besides, you have to add mlxlink to subdirs of the main makefile of mstflint

**No need to add mlxlink to the main makefile, I added it in adabe tools (ADABE_TOOLS="adb_parser mlxreg mlxlink") in configure.ac**
**because it depends on adb_parser and mlxreg.**

**Pull request merged**